### ISSUE TITLE
coll/han does not compile with nvcc 22.X

### ISSUE THREAD

**Issue by hppritcha** [open-mpi/ompi#11384]
In trying to build Open MPI main and 5.0.0rc10 with nvidia compilers (22.7), an internal compiler error is hit when compiling coll_han_component.c

```
NVC++-F-0000-Internal compiler error. unhandled type      44 
```

I do not have access to a newer compiler. 

The workaround for now is to build Open MPI without the HAN collective component.  Add ```--enable-mca-no-build=coll-han``` to the Open MPI configure line.

**Comment by bosilca** [comment]
I'm not sure how I can help here, we don't even know what type or around what line the issue is located. @hppritcha do you have any additional information ?

**Comment by hppritcha** [comment]
The compiler doesn't seem to like the COMPONENT_T component at ~line 250 in the file.  No idea why that is.  Nothing illegal looking about its definition.  It seems to be a little more complex than this structure alone though, because a simple standalone test declaring that structure and using it doesn't exhibit the compiler failure.

**Comment by bosilca** [comment]
Try to change it from COMPONENT_T to uint32_t. I bet it does not like to use an enum as an iterator in the loop at line 466.

**Comment by hppritcha** [comment]
i'll give that a try.

**Comment by hppritcha** [comment]
unfortunately that suggestion didn't change the result - compiler still fails with an internal error.

**Comment by hppritcha** [comment]
okay the problem for nvcc was initializing ```mca_coll_han_component``` without setting
dynamic_rules_filename to NULL.  apparently its static initializer code is kind of iffy.

**Comment by hppritcha** [comment]
closed vida #11413 
and #11416
